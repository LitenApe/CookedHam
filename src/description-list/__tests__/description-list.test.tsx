import React from "react";
import TestRenderer from "react-test-renderer";

import { DescriptionList } from "../index";

describe("DescriptionList", () => {
  it("renders without crashing", () => {
    TestRenderer.create(<DescriptionList items={[]} />);
  });

  it("renders a title and description each row", () => {
    const dl = TestRenderer.create(<DescriptionList items={[
      { label: "label-1", description: "description-1", },
      { label: "label-2", description: "description-2", },
      { label: "label-3", description: "description-3", },
      { label: "label-4", description: "description-4", }
    ]}
    />);

    expect(dl).toMatchSnapshot();
  });

  it("renders multiple title and a description each row", () => {
    const dl = TestRenderer.create(<DescriptionList items={[
      { label: ["label-1a", "label-1b", "label1c"], description: "description-1", },
      { label: ["label-2a", "label-2b", "label2c"], description: "description-2", },
      { label: ["label-3a", "label-3b", "label3c"], description: "description-3", },
      { label: ["label-4a", "label-4b", "label4c"], description: "description-4", }
    ]}
    />);

    expect(dl).toMatchSnapshot();
  });

  it("renders a title and multiple description each row", () => {
    const dl = TestRenderer.create(<DescriptionList items={[
      { label: "label-1", description: ["description-1a", "description-1b", "description-1c"], },
      { label: "label-2", description: ["description-2a", "description-2b", "description-2c"], },
      { label: "label-3", description: ["description-3a", "description-3b", "description-3c"], },
      { label: "label-4", description: ["description-4a", "description-4b", "description-4c"], }
    ]}
    />);

    expect(dl).toMatchSnapshot();
  });
});
