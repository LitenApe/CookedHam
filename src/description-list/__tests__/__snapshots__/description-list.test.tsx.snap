// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DescriptionList renders a title and description each row 1`] = `
<dl
  className="ch-description-list"
>
  <div>
    <dt>
      label-1
    </dt>
    <dd>
      description-1
    </dd>
  </div>
  <div>
    <dt>
      label-2
    </dt>
    <dd>
      description-2
    </dd>
  </div>
  <div>
    <dt>
      label-3
    </dt>
    <dd>
      description-3
    </dd>
  </div>
  <div>
    <dt>
      label-4
    </dt>
    <dd>
      description-4
    </dd>
  </div>
</dl>
`;

exports[`DescriptionList renders a title and multiple description each row 1`] = `
<dl
  className="ch-description-list"
>
  <div>
    <dt>
      label-1
    </dt>
    <dd>
      description-1a
    </dd>
    <dd>
      description-1b
    </dd>
    <dd>
      description-1c
    </dd>
  </div>
  <div>
    <dt>
      label-2
    </dt>
    <dd>
      description-2a
    </dd>
    <dd>
      description-2b
    </dd>
    <dd>
      description-2c
    </dd>
  </div>
  <div>
    <dt>
      label-3
    </dt>
    <dd>
      description-3a
    </dd>
    <dd>
      description-3b
    </dd>
    <dd>
      description-3c
    </dd>
  </div>
  <div>
    <dt>
      label-4
    </dt>
    <dd>
      description-4a
    </dd>
    <dd>
      description-4b
    </dd>
    <dd>
      description-4c
    </dd>
  </div>
</dl>
`;

exports[`DescriptionList renders multiple title and a description each row 1`] = `
<dl
  className="ch-description-list"
>
  <div>
    <dt>
      label-1a
    </dt>
    <dt>
      label-1b
    </dt>
    <dt>
      label1c
    </dt>
    <dd>
      description-1
    </dd>
  </div>
  <div>
    <dt>
      label-2a
    </dt>
    <dt>
      label-2b
    </dt>
    <dt>
      label2c
    </dt>
    <dd>
      description-2
    </dd>
  </div>
  <div>
    <dt>
      label-3a
    </dt>
    <dt>
      label-3b
    </dt>
    <dt>
      label3c
    </dt>
    <dd>
      description-3
    </dd>
  </div>
  <div>
    <dt>
      label-4a
    </dt>
    <dt>
      label-4b
    </dt>
    <dt>
      label4c
    </dt>
    <dd>
      description-4
    </dd>
  </div>
</dl>
`;
